chore: deprecate and block 'access_control' field in Team API

zlwaterfield opened the pull request:
## Changes

Adds validation to the TeamSerializer to reject usage of the deprecated 'access_control' field, raising a validation error and capturing the event for analytics. Removes legacy permission checks for 'access_control' in team attribute validation. Updates tests to ensure the field is blocked and proper errors are returned.

## How did you test this code?

Added tests

## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [ ] No docs needed for this change

---

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

zlwaterfield committed:
Deprecate and block 'access_control' field in Team API

Adds validation to the TeamSerializer to reject usage of the deprecated 'access_control' field, raising a validation error and capturing the event for analytics. Removes legacy permission checks for 'access_control' in team attribute validation. Updates tests to ensure the field is blocked and proper errors are returned.

joshsny commented on posthog/api/team.py:
```diff
@@ -458,6 +457,31 @@ def validate_session_replay_ai_summary_config(value: dict | None) -> dict | None
+                },
+            )
+
+            raise exceptions.ValidationError(
```
low on context, but just checking we are ready to raise an exception here? Do we want to add the analytics first then do a PR to raise later, or do we expect no one to be using it at all?

joshsny reviewed (approved):
LGTM, assuming we're ready to raise the exception and expect that no one will be trying to use it

zlwaterfield committed:
Merge branch 'master' into zach/access-control/deprecate-team-access-control

zlwaterfield committed:
Merge branch 'master' into zach/access-control/deprecate-team-access-control

zlwaterfield committed:
Merge branch 'master' into zach/access-control/deprecate-team-access-control

zlwaterfield committed:
Merge branch 'master' into zach/access-control/deprecate-team-access-control

zlwaterfield committed:
Add access control validation and update tests

Introduces validate_access_control to ProjectBackwardCompatSerializer, delegating to TeamSerializer. Updates test_team.py to ensure organization membership level is set to ADMIN and available product features are configured before environment-related access control tests.

zlwaterfield committed:
Remove old access control tests

zlwaterfield committed:
Update access_control validation and test setup

Refactored access_control validation methods in ProjectBackwardCompatSerializer and TeamSerializer to consistently return None and updated type hints. Simplified available_product_features setup in team API tests by removing redundant list expansion.

zlwaterfield committed:
Add ENVIRONMENTS feature to organization test setup

Included the ENVIRONMENTS feature in the organization's available product features within the team API test factory to ensure tests cover this feature.
